mate-session/gsm-manager.c

```diff
@@ -1178,26 +1170,16 @@ manager_attempt_hibernate (GsmManager *manager)
 
                 gsm_systemd_attempt_hibernate (systemd);
         }
-#endif
-#if defined(HAVE_SYSTEMD) && defined(HAVE_UPOWER_HIBERNATE_SUSPEND)
-        else {
-#endif
-#ifdef HAVE_UPOWER_HIBERNATE_SUSPEND
-        gboolean can_hibernate = up_client_get_can_hibernate (manager->priv->up_client);
+#else
```

Comment:
I might have missed something, but this looks like a transition to compile-time only choice. Since other parts of this file have both compile-time and runtime choice between systemd and consolekit, I suggest doing the same here.

The same applies to the suspend code changes a few lines below.